src/components/VmDetails/cards/NicsCard/NicEditor.js

```diff
@@ -0,0 +1,303 @@
+
+          <Form horizontal>
+            <FormGroup controlId='nic-name' disabled={false} className='required'>
+              <LabelCol sm={3}>
+                { msg.nicEditorNameLabel() }
+              </LabelCol>
+              <Col sm={9}>
+                <FormControl
+                  type='text'
+                  defaultValue={this.state.name}
+                  onChange={this.changeName}
+                />
+              </Col>
+            </FormGroup>
+
+            <FormGroup controlId='nic-vnic-profile' disabled={false} className='required'>
+              <LabelCol sm={3}>
+                { msg.vnicProfile() }
+              </LabelCol>
+              <Col sm={9}>
+                <SelectBox
+                  idPrefix='nic-vnic-profiles-select'
+                  items={vnicList}
+                  selected={this.state.vnicProfileId}
+                  onChange={this.changeVnicProfile}
+                />
+              </Col>
+            </FormGroup>
+
+            <ExpandCollapseSection id='AdvancedOptions'>
+              <FormGroup controlId='nic-type'>
+                <LabelCol sm={3}>
+                  { msg.nicEditorInterfaceLabel() }
+                  { !canChangeInterface &&
+                    <FieldLevelHelp
+                      inline
+                      content={msg.nicEditorInterfaceCantEditHelp()}
+                      buttonClass={style['field-help']}
+                    />
+                  }
+                </LabelCol>
+                <Col sm={9}>
+                  { !canChangeInterface &&
+                    <div className={style['editor-value-text']}>
+                      { nicInterface ? nicInterface.value : 'N/A' }
+                    </div>
+                  }
+                  { canChangeInterface &&
+                    <SelectBox
+                      idPrefix='nic-type-select'
+                      items={NIC_INTERFACES}
+                      selected={NIC_INTERFACE_DEFAULT}
+                      onChange={this.changeInterface}
+                    />
+                  }
+                </Col>
+              </FormGroup>
+
+              <FormGroup controlId='nic-link-state'>
+                <LabelCol sm={3}>
+                  { msg.nicEditorLinkStateLabel() }
+                </LabelCol>
+                <Col sm={9}>
+                  <Radio
+                    name='nic-link-state-group'
+                    defaultChecked={this.state.linked}
+                    onChange={() => { this.changeLinked(true) }}
+                  >
+                    Up <NicLinkStateIcon linkState idSuffix='up' showTooltip={false} />
+                  </Radio>
+                  <Radio
+                    name='nic-link-state-group'
+                    defaultChecked={!this.state.linked}
+                    onChange={() => { this.changeLinked(false) }}
+                  >
+                    Down <NicLinkStateIcon linkState={false} idSuffix='down' showTooltip={false} />
```

Comment:
Ah, not quite.  `linkState` is set `PropTypes.bool.isRequired` and will throw a horrible error killing the entire app if it isn't explicitly set (since not setting it means `undefined` not `false`, and while `undefined` is falsy, it is not `false`).

Net result, in `NicLinkStateIcon`, I dropped the `isRequired` and defaulted `linkState` to `false`.  Then I can drop the `linkState={false}` bit. :-)